sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py

```diff
@@ -293,6 +296,20 @@ async def delete_entity(
                 :dedent: 8
                 :caption: Adding an entity to a Table
         """
+        try:
+            entity = kwargs.pop('entity', None)
+            if not entity:
+                entity = args[0]
+            partition_key = entity['PartitionKey']
+            row_key = entity['RowKey']
+        except (TypeError, IndexError):
+            partition_key = kwargs.pop('partition_key', None)
+            if not partition_key:
+                partition_key = args[0]
+            row_key = kwargs.pop('row_key', None) or args[1]
```

Comment:
You did the `or` in the async version too ;)

Reply:
🤦‍♂️